interval.hh

```diff
@@ -154,6 +154,33 @@ public:
         }
         return false;
     }
+    // the other inverval is before this interval (works only for non wrapped intervals)
+    // Comparator must define a total ordering on T.
+    bool before(const wrapping_interval<T>& o, IntervalComparatorFor<T> auto&& cmp) const {
```

Comment:
I know you followed the other overload, but it leads to "a.before(b)" == "b is before a". I'm sure it will catch the unwary reader.

I suggest std::partial_ordering compare() (and define overlapping ranges as equivalent). Or find a new name, this one is confusing.

Reply:
Yes, it was already confusing with points, it is even more confusing with intervals.

I would change to `after()`, but we already have an `after()` method which is just as confusing. Also such a swap rename would get people who are already use to the current confusing naming.

I can try prepending and appending something to `before`, but there is really no winning here.

Reply:
Renamed to `other_is_before()` in v6